properly conjugate visibilities for cross-corr get_data

adampbeardsley opened the pull request:
Addresses Issue #398

bhazelton commented on pyuvdata/uvdata.py:
```diff
@@ -2159,28 +2159,48 @@ def _key2inds(self, key):
             blt_ind2: numpy array with blt indices for conjugate antenna pair.
-            pol_ind: numpy array with polarization indices
+                      Note if a cross-pol baseline is requested, the polarization will
+                      also be reversed. e.g. (1, 2, 'xy') -> conj(2, 1, 'yx')
```
I find this comment confusing. I'm not sure when I'm getting reversed polarizations and whether that means they match what I asked for or not.

adampbeardsley commented:
@bhazelton I think I've addressed your comments, though I'm not sure how clear my doc string is.. I'm having a hard time articulating it right. Let me know what you think.
I actually changed the behavior for a couple of the places you requests tests for. If the user only supplies a polarization key, I don't think we should be returning the conjugate pol. So now if you ask for 'xy' you only get 'xy', and not conj('yx').

jsdillon commented:
Isn't there a much simpler way to do this? 

```
from pyuvdata.utils import polstr2num

def conj_pol(pol):
    '''Given V_ij^(pol), return the polarization of V_ji^(conj pol) such
    (V_ji^(conj pol))* = V_ij^(pol). This means xy -> yx and yx -> xy, but
    psuedo-Stokes visibilities are unaffected. Case is left unmodified.'''
    if polstr2num(pol) > 0:  # this includes Stokes and pseudo-Stokes visibilities
        return pol
    else:
        return pol[::-1]
```
